numpy/polynomial/chebyshev.py

```diff
@@ -1830,6 +1831,71 @@ def chebcompanion(c):
     return mat
 
 
+def chebinterp(func, deg, x=None, args=(), window=None):
+    """Approximate a function using Chebyshev polynomial interpolation.
+
+    Return the Chebyshev series instance that interpolates a given function.
+
+    Parameters
+    ----------
+    func : function
+        The function to be approximated. It must be a function of a single
+        variable of the form f(x,a,b,c...), where a,b,c... are extra arguments
+        that can be passed in the `args` parameter.
+    deg : int or 1-D array_like
+        Degree(s) of the fitting polynomials. If `deg` is a single
+        integer all terms up to and including the `deg`'th term are included in
+        the interpolation.
+    x : array_like, shape (M,), optional
+        Defines window of interpolation. The domain of `x` is used as the
+        approximation window if `window` is not given.
+    args : tuple, optional
+        Extra arguments to be used in the function call.
+    window : [beg, end], optional
+        Window to use for the returned series.
+
+    Returns
+    -------
+    new_series : series
+        A series that represents the polynomial interpolation
+        to the function.
+
+    Notes
+    -----
+    A low-pass filter derived from Chebyshev approximation has an
```

Comment:
Note the Chebyshev filters of type I and II are more complicated than simple Chebyshev polynomials. Filters really belong in the scipy signal processing package.